### PR TITLE
Add CRUD UI for rules

### PR DESCRIPTION
**Pull request by dmitriim**

### CODE REVIEW

File: classes/output/renderer.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     * Render rules table.
+     *
+     * @param managerules $rules
+     * @return string
+     */
+    public function render_managerules(managerules $rules): string {
+        $html = '';
+        ob_start();
+        $rules->out($rules->pagesize, true);
+        $html .= ob_get_contents();
```
Comment (cameron1729) [inline classes/output/renderer.php:42]: We don't concatenate anything else here, so could we just do `$html = ob_get_contents();` ?

File: delete.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Delete
```
Comment (cameron1729) [inline delete.php:18]: Needs a full stop? And maybe a bit more of a description.

Comment (dmitriim) [reply delete.php]: fixed

File: classes/output/renderer.php (hunk 1 above)
Comment (dmitriim) [reply classes/output/renderer.php]: fixed

File: classes/helper.php [hunk 3]
```diff
@@ -101,4 +106,51 @@ protected static function is_valid_rule_data(\stdClass $formdata): bool {
...
+     * @return \moodle_url
+     */
+    public static function build_rule_edit_url(rule $rule): moodle_url {
+        return new moodle_url('/admin/tool/cohortmanager/edit.php', ['ruleid' => $rule->get('id')]);
+    }
+
+    /**
+     * Builds rule delete URL.
+     *
+     * @param \tool_cohortmanager\rule $rule Rule instance.
```
[the remark this answers is not in the document]
Comment (dmitriim) [reply classes/helper.php]: fixed

File: classes/helper.php [hunk 4]
```diff
@@ -101,4 +106,51 @@ protected static function is_valid_rule_data(\stdClass $formdata): bool {
...
+        return $cohorts;
+    }
+
+    /**
+     * Builds rule edit URL.
+     *
+     * @param \tool_cohortmanager\rule $rule Rule instance.
+     * @return \moodle_url
+     */
+    public static function build_rule_edit_url(rule $rule): moodle_url {
```
[the remark this answers is not in the document]
Comment (dmitriim) [reply classes/helper.php:130]: fixed

**Pull request merged**